### PR TITLE
🐛Autoscaling: Ensure buffer machine type remains the same over the app session

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
This PR fixes the issue where a different machine type could replace the intended buffer machine type.
- fixes https://github.com/ITISFoundation/osparc-simcore/issues/5453
@mguidon this should allow setting multiple machine types, and still keeping the correct type of buffer machines (iSeg use-case)
## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by bisgaard-itis — APPROVED** [review]
Thanks, looks good

File: services/autoscaling/src/simcore_service_autoscaling/utils/auto_scaling_core.py [hunk 1]
```diff
@@ -148,3 +148,34 @@ def find_selected_instance_type_for_task(
...
     return selected_instance
+
+
+def get_machine_buffer_type(
+    available_ec2_types: list[EC2InstanceType],
+) -> EC2InstanceType:
+    return available_ec2_types[0]
+
+
+DRAINED_NODES = list[AssociatedInstance]
```
Comment (pcrespov) [inline services/autoscaling/src/simcore_service_autoscaling/utils/auto_scaling_core.py:159]: `TypeAlias`
Why type alias is written as a constant here?

**Review by pcrespov — APPROVED** [review]

File: services/autoscaling/src/simcore_service_autoscaling/utils/auto_scaling_core.py [hunk 2]
```diff
@@ -148,3 +148,34 @@ def find_selected_instance_type_for_task(
         raise Ec2InstanceInvalidError(msg=msg)
 
     return selected_instance
+
+
+def get_machine_buffer_type(
+    available_ec2_types: list[EC2InstanceType],
+) -> EC2InstanceType:
+    return available_ec2_types[0]
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/autoscaling/src/simcore_service_autoscaling/utils/auto_scaling_core.py:156]: yes

File: services/autoscaling/src/simcore_service_autoscaling/utils/auto_scaling_core.py (hunk 1 above)
Comment (sanderegg) [reply services/autoscaling/src/simcore_service_autoscaling/utils/auto_scaling_core.py:159]: yes, added assert

**Pull request merged**